File: 준영/프로그래머스/LV2/행렬 테두리 회전하기.js
function solution(rows, columns, queries) {
  var answer = []
  var init_arr = Array.from(Array(rows), () => new Array(columns).fill(0))
  for (let i = 1; i <= rows; i++) {
    for (let j = 1; j <= columns; j++) {
      init_arr[i-1][j-1] = (i-1)*columns + j
    }
  }
  for (const query of queries) {
    var temp = Array.from(Array(rows), () => new Array(columns).fill(0))
    var arr = []
    var start_x = query[0]
    var start_y = query[1]
    var end_x = query[2]
    var end_y = query[3]

    for (let j = start_y; j < end_y; j++) {
      temp[start_x-1][j] = init_arr[start_x-1][j-1]
      arr.push(temp[start_x-1][j])
    }
    
    for (let i = start_x; i < end_x; i++) {
      temp[i][end_y-1] = init_arr[i-1][end_y-1]
      arr.push(temp[i][end_y-1])
    }
    
    for (let j = start_y; j < end_y; j++) {
      temp[end_x-1][j-1] = init_arr[end_x-1][j]
      arr.push(temp[end_x-1][j-1])
    }
    
    for (let i = start_x; i < end_x; i++) {
      temp[i-1][start_y-1] = init_arr[i][start_y-1]
      arr.push(temp[i-1][start_y-1])
    }

    for (let i = 0; i < rows; i++) {
      for (let j = 0; j < columns; j++) {
        if (temp[i][j] != 0) {
          init_arr[i][j] = temp[i][j]
        }
      }
    }
    answer.push(Math.min(...arr))
  }

  console.log(answer)
  return answer
}

rows = 6
columns = 6
queries = [[2,2,5,4],[3,3,6,6],[5,1,6,3]]
solution(rows, columns, queries)
rows = 100
columns = 97
queries = [[1,1,100,97]]	
solution(rows, columns, queries)

File: 준영/프로그래머스/LV2/행렬 테두리 회전하기.py
def solution(rows, columns, queries):
    answer = []
    init_arr = [[0 for _ in range(columns)] for _ in range(rows)]

    for i in range(1, rows+1):
        for j in range(1, columns+1):
            init_arr[i-1][j-1] = (i-1)*columns + j

    for query in queries:
        temp = [[0 for _ in range(columns)] for _ in range(rows)]
        arr = []
        start_x = query[0]
        start_y = query[1]
        end_x = query[2]
        end_y = query[3]

        for j in range(start_y, end_y):
            temp[start_x-1][j] = init_arr[start_x-1][j-1]
            arr.append(temp[start_x-1][j])

        for i in range(start_x, end_x):
            temp[i][end_y-1] = init_arr[i-1][end_y-1]
            arr.append(temp[i][end_y-1])

        for j in range(start_y, end_y):
            temp[end_x-1][j-1] = init_arr[end_x-1][j]
            arr.append(temp[end_x-1][j-1])

        for i in range(start_x, end_x):
            temp[i-1][start_y-1] = init_arr[i][start_y-1]
            arr.append(temp[i-1][start_y-1])

        for i in range(rows):
            for j in range(columns):
                if temp[i][j] != 0:
                    init_arr[i][j] = temp[i][j]
        
        answer.append(min(arr))

    return answer

rows = 6
columns = 6
queries = [[2,2,5,4],[3,3,6,6],[5,1,6,3]]
print(solution(rows, columns, queries))
rows = 3
columns = 3
queries = [[1,1,2,2],[1,2,2,3],[2,1,3,2],[2,2,3,3]]	
print(solution(rows, columns, queries))
rows = 100
columns = 97
queries = [[1,1,100,97]]	
print(solution(rows, columns, queries))